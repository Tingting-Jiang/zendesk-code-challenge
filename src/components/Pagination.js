import React, { useState } from 'react'


const Pagination = ({ticketsPerPage, totalTickets, paginate}) =>{
    const pageNumbers = [];
    for (let i = 1; i <= Math.ceil(totalTickets / ticketsPerPage ); i++) {
        pageNumbers.push(i);
    };
    
    const [active, setActive] = useState(1);
  
    return (

            <ul className="pagination nav nav-pills align-items-center ">
                {pageNumbers.map(number => (
                    <li key={number}
                        className={`page-item
                    ${active === number ? "active" : ""}`}   >
                        <a onClick={() => {
                            paginate(number);
                            setActive(number);
                            
                        }}
                            href="!#"
                           className="page-link"
                           id="change-page"
                      >
                            {number}
                        </a>
                    </li>
                ))}
            </ul>

    )
};
export default Pagination;


