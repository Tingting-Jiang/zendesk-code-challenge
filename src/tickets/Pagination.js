import React, { useState } from 'react'


const Pagination = ({ticketsPerPage, totalTickets, paginate}) =>{
    const pageNumbers = [];
    for (let i = 1; i <= Math.ceil(totalTickets / ticketsPerPage ); i++) {
        pageNumbers.push(i);
    };
    
    

    
    return (
        <nav className="nav-pills">
            <ul className="pagination">
                {pageNumbers.map(number => (
                    <li key={number} className="page-item"   >
                        <a onClick={() => {
                            paginate(number);
                            
                        }}
                            href="!#" className="page-link "
                      >
                            {number}
                        </a>
                    </li>
                ))}
            </ul>
        </nav>
    )
};
export default Pagination;


