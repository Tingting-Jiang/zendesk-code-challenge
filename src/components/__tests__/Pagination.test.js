import React from 'react';
import Pagination from '../Pagination';
import {shallow, mount, render } from"enzyme";
import renderer from "react-test-renderer";
import { cleanup } from '@testing-library/react'





describe("<Pagination/>", () =>{
    // let wrapper;
    // beforeEach(() => {
    //     wrapper = shallow(<Pagination />);
    // })
    //
    afterAll( () => {
        cleanup();
    })
    
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState')
    useStateSpy.mockImplementation((init) => [init, setState]);
    
    
    it("render the Pagination without data", () => {
        const wrapper = shallow(<Pagination />);
        // const tree = renderer.create(<Pagination />).toJSON();
        expect(wrapper).toMatchSnapshot();

    });
    
    it("render Pagination with data", () => {
        const ticketPerPage = 2;
        const totalTickets = 3;
        const wrapper = shallow(<Pagination
            ticketsPerPage={ticketPerPage}
            totalTickets={totalTickets} />);
        const item = Math.ceil(totalTickets / ticketPerPage);
        expect(wrapper.find('ul').children()).toHaveLength(item);
    })
    
    
    
    // it("render Pagination with data", () => {
    //     const ticketPerPage = 3;
    //     const totalTickets = 4;
    //     const paginate = jest.fn();
    //
    //     const wrapper = shallow(<Pagination
    //         ticketsPerPage={ticketPerPage}
    //         totalTickets={totalTickets}
    //         paginate={paginate}/>);
    //     wrapper.find('#change-page').props().onClick();
    //     // wrapper.find('#change-page').simulate('click');
    //     expect(paginate).toHaveBeenCalled();
    // })
    
    
    
    
    
    
    
});

