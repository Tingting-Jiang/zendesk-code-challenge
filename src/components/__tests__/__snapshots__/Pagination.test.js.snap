// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> render the Pagination without data 1`] = `ShallowWrapper {}`;
